cppcache/src/CacheXmlCreation.cpp

```diff
@@ -34,37 +35,21 @@ void CacheXmlCreation::addPool(std::shared_ptr<PoolXmlCreation> pool) {
   pools.push_back(pool);
 }
 
-void CacheXmlCreation::create(Cache* cache) {
-  m_cache = cache;
-  m_cache->m_cacheImpl->setPdxIgnoreUnreadFields(m_pdxIgnoreUnreadFields);
-  m_cache->m_cacheImpl->setPdxReadSerialized(m_readPdxSerialized);
+void CacheXmlCreation::create(Cache& cache) {
   // Create any pools before creating any regions.
-
-  for (const auto& pool : pools) {
-    pool->create();
-  }
-
-  for (const auto& rootRegion : rootRegions) {
-    rootRegion->createRoot(cache);
-  }
-}
-
-void CacheXmlCreation::setPdxIgnoreUnreadField(bool ignore) {
-  // m_cache->m_cacheImpl->setPdxIgnoreUnreadFields(ignore);
-  m_pdxIgnoreUnreadFields = ignore;
+  std::for_each(std::begin(pools), std::end(pools),
+                std::bind(&PoolXmlCreation::create,
+                          std::bind(&std::shared_ptr<PoolXmlCreation>::get,
+                                    std::placeholders::_1)));
```

Comment:
Hmm... Does this modern C++ style make this any more readable?

Would dereferencing the `shared_ptr`s as local variables prior to these calls help with readability?
Would a lambda rather than `std::bind` read easier?

Without the side by side diff I honestly can't say I understood what is going on here without unwinding all the algorithm magic going on here.